File: backend/services/tenant_service.py

from repositories.tenant_repository import TenantRepository

class TenantService:
    def get_all(self):
        return TenantRepository().get_all()

    def get_by_id(self, tenant_id):
        return TenantRepository().get_by_id(tenant_id)

    def create(self, tenant_code, tenant_abbreviation, tenant_name, db_name):
        return TenantRepository().create(tenant_code, tenant_abbreviation, tenant_name, db_name)

    def update(self, tenant_id, tenant_code, tenant_abbreviation, tenant_name, db_name):
        return TenantRepository().update(tenant_id, tenant_code, tenant_abbreviation, tenant_name, db_name)

    def set_active(self, tenant_id, is_active):
        return TenantRepository().set_active(tenant_id, is_active)

File: frontend/src/services/tenantService.ts
import { api } from './apiClient'
import type { Tenant, TenantInput } from '../types/tenant'

export const tenantService = {
  list: () => api.get<Tenant[]>('/api/tenants'),
  getById: (tenantId: string) => api.get<Tenant>(`/api/tenants/${tenantId}`),
  create: (input: TenantInput) => api.post<Tenant>('/api/tenants', input),
  update: (tenantId: string, input: TenantInput) =>
    api.put<Tenant>(`/api/tenants/${tenantId}`, input),
  setStatus: (tenantId: string, isActive: boolean) =>
    api.patch<Tenant>(`/api/tenants/${tenantId}/status`, { is_active: isActive }),
}
